src/echoclient/EchoClient.php

```diff
@@ -429,6 +423,58 @@ protected function getHeaders($noCache = false)
         return $headers;
     }
 
+    /**
+     * Sends json encoded events data to echo
+     *
+     * @param string $eventsData The json encoded events data to send
+     * @return bool True if successful
+     * @throws EchoHttpException If the server responded with an error
+     * @throws EchoCouldNotSendException If we were unable to send data to the Echo server
+     */
+    protected function sendJsonEventDataToEcho($eventsData)
+    {
+        $baseUrl = $this->getBaseUrl();
+
+        if (!$baseUrl) {
+            // fail silently when creating events, should not stop user interaction as echo events are collected on a best-endeavours basis
+            $this->getLogger()->warning('Echo server is not defined (missing ECHO_HOST define), not sending events');
+            return false;
+        }
+
+        $eventUrl = $baseUrl . '/events';
+
+        try {
+            $client = $this->getHttpClient();
```

Comment:
does this one and the one below throw a exception? Would it be clearer if both of these lines was out side of the try catch as we don't expect a exception to occur?

Reply:
i'm not 100% sure is the honest answer, I think the only line that throws the exception is `->send()`

Reply:
I've re-organised the logic in here to make it clearer.